### PR TITLE
feat(excel-exporter): add an option to export headers when there's no data

### LINKED ISSUE #10702: Excel export : headers not displayed when no data is present on the grid
We are unable to see headers for a given Excel export in case there is no data present on the grid. An empty Excel file is exported without any headers. Our expected output is an Excel file with column headers according to those present on the grid. 
Is there a way to achieve this functionality?

Component : IgxExcelExporterService

### PR DESCRIPTION
**Pull request by onlyexeption**
Closes #10702

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/services/excel/excel-files.ts [hunk 1]
```diff
@@ -205,16 +210,27 @@ export class WorksheetFile implements IExcelFile {
                 }
             }
 
-            this.processDataRecordsAsync(worksheetData, (rows) => {
-                sheetData += rows;
+            if (worksheetData.isEmpty) {
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/services/excel/excel-files.ts:213]: Is this branch needed? Can't processDataRecordsAsync handle the case properly?

File: projects/igniteui-angular/src/lib/services/excel/worksheet-data.ts [hunk 2]
```diff
@@ -47,14 +47,24 @@ export class WorksheetData {
     }
 
     private initializeData() {
+        this._dataDictionary = new WorksheetDataDictionary(this.columnCount, this.options.columnWidth, this.columnWidths);
+
         if (!this._data || this._data.length === 0) {
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/services/excel/worksheet-data.ts:52]: There's some repetition here which we need to review.

**Review by gedinakova — CHANGES_REQUESTED** [review]
Add the new feature to CHANGELOG.

File: projects/igniteui-angular/src/lib/services/excel/excel-exporter-grid.spec.ts [hunk 3]
```diff
@@ -902,12 +906,13 @@ describe('Excel Exporter', () => {
             await exportAndVerify(hGrid, options, actualData.exportMultiColumnHeadersDataWithSkippedParentMCH);
         });
 
-        it('should export empty file when all parent multi column headers are skipped', async () => {
+        fit('should export empty file when all parent multi column headers are skipped and alwaysExportHeaders is false', async () => {
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/services/excel/excel-exporter-grid.spec.ts:909]: Remove the focus from this test.

**Review by gedinakova — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/services/excel/excel-exporter-grid.spec.ts [hunk 4]
```diff
@@ -1173,7 +1173,7 @@ describe('Excel Exporter', () => {
             await exportAndVerify(grid, options, actualData.exportCollapsedAndExpandedMultiColumnHeadersData, false);
         });
 
-        it('should respect ignoreMultiColumnHeaders when set to true', async () => {
+        fit('should respect ignoreMultiColumnHeaders when set to true', async () => {
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/services/excel/excel-exporter-grid.spec.ts:1176]: Remove this f.

**Review by gedinakova — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/services/excel/excel-files.ts [hunk 5]
```diff
@@ -404,7 +406,8 @@ export class WorkbookFile implements IExcelFile {
  */
 export class ContentTypesFile implements IExcelFile {
     public writeElement(folder: JSZip, worksheetData: WorksheetData) {
-        folder.file('[Content_Types].xml', ExcelStrings.getContentTypesXML(!worksheetData.isEmpty, worksheetData.options.exportAsTable));
+        const hasSharedStrings = !worksheetData.isEmpty || (worksheetData.isEmpty && worksheetData.options.alwaysExportHeaders);
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/services/excel/excel-files.ts:409]: Do we need the "(worksheetData.isEmpty &&" portion? If alwaysExportHeaders is true, we will have shared strings with or without data, right?

File: projects/igniteui-angular/src/lib/services/excel/excel-folders.ts [hunk 6]
```diff
@@ -70,7 +70,7 @@ export class XLExcelFolder implements IExcelFolder {
             ExcelFileTypes.WorkbookFile
         ];
 
-        if (!data.isEmpty) {
+        if (!data.isEmpty || (data.isEmpty && data.options.alwaysExportHeaders)) {
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/services/excel/excel-folders.ts:73]: Same as above.

File: projects/igniteui-angular/src/lib/services/exporter-common/exporter-options-base.ts [hunk 7]
```diff
@@ -78,6 +78,17 @@ export abstract class IgxExporterOptionsBase {
...
+     * this.exportOptions.alwaysExportHeaders = false;
+     * ```
+     *
+     * @memberof IgxExporterOptionsBase
+     */
+     public alwaysExportHeaders = true;
+
     private _fileName: string;
 
     constructor(fileName: string, protected _fileExtension: string) {
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/services/exporter-common/exporter-options-base.ts:94]: Add the feature to the CHANGELOG.md.

**Review by gedinakova — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/services/excel/excel-files.ts (hunk 5 above)
Comment (onlyexeption) [reply projects/igniteui-angular/src/lib/services/excel/excel-files.ts:409]: yes but we shouldn't have shared strings if alwaysExportHeaders is false and data is empty

Comment (gedinakova) [reply projects/igniteui-angular/src/lib/services/excel/excel-files.ts:409]: The result from !worksheetData.isEmpty || (worksheetData.isEmpty && worksheetData.options.alwaysExportHeaders) is the same as !worksheetData.isEmpty || worksheetData.options.alwaysExportHeaders.

**Review by gedinakova — CHANGES_REQUESTED** [review]

**Review by gedinakova — APPROVED** [review]

**Pull request merged**